### ISSUE TITLE
Explicit Gitea support

### ISSUE THREAD

**Issue by ericek111** [repman-io/repman#309]
I haven't found a single mention of Gitea anywhere. For a newcomer (such as myself), this can be discouraging. Maybe we should offer documentation (and options, where applicable) specific to Gitea?

**Comment by akondas** [comment]
Hey @ericek111, thanks for feedback. Of course we can add such information.
Did you manage to connect Repman with Gitea and whether any changes were needed in the source code?

**Comment by ericek111** [comment]
So far I've only managed to add support for OAuth registration and login using Gitea as the provider. It's my first time working with Symphony, so I'm not all that proficient. Here's my fork, it still needs functional and integration tests: https://github.com/ericek111/repman/tree/feature-gitea

I'm planing on submitting a PR once the tests are done. Suggestions and changes are welcome, of course. The next order of action will be adding support for private Gitea repositories. Syncing hangs using both HTTP and SSH for VCS repos. 

I'm thinking that a separate option for Gitea would be benificial, though Repman should support private SSH Git repos. Maybe a field to input the user's authorised private key for VCS repos would be the best?

**Comment by akondas** [comment]
Yes, we have already thought about the option to be able to provide your own ssh key.

I would see it as a separate tab where you can add your keys, and then on the package adding card it would be possible to select and authenticate a given key. Here is a simple visualization of how I imagine it:

![Screenshot from 2020-10-22 13-13-01](https://user-images.githubusercontent.com/8239917/96864221-60e97900-1468-11eb-8d11-e23a793a0fd0.png)

**Comment by akondas** [comment]
I have created a separate task for ssh keys so as not to clutter this thread.

**Comment by mihaileu** [comment]
Hi, we have a self hosted Stash Bitbucket, the repos are private and we are having difficulties in adding them in repman.   
Is there an existing solution for this ? @akondas

**Comment by akondas** [comment]
What's the problem? (i don't have experience with Stash Bitbucket)

**Comment by mihaileu** [comment]
We are using Atlassian Bitbucket v6.2.0 hosted on a custom domain. 

We need to authenticate in order to have access to repos. In the web interface there's no option for adding a ssh key or other option to authenticate with a self hosted bitbucket.  

 Currently we will try to add a key when starting the docker container in order to be able to pull private repository from this custom domain.
